### ISSUE TITLE
Migrate from using bind_to_port=false to filter chains.

### ISSUE THREAD

**Issue by PiotrSikora** [istio/istio#10533]
Envoy removed support for `bind_to_port` (https://github.com/envoyproxy/envoy/pull/5288), which has been marked as deprecated_v1 since 2017-08-15 (https://github.com/envoyproxy/data-plane-api/commit/86de1f257534b931232629b855a3ddbabda7bdc5), and announced to be removed on 2018-03-15 (https://groups.google.com/forum/#!topic/envoy-announce/Lb1QZcSclGQ), so it was long time coming, but we cannot upgrade Envoy in Istio Proxy now...

The feature was superseded by filter chains, and so we should migrate from using multiple listeners with `bind_to_port=false` to a single listener with multiple filter chains.

cc @duderino @costinm @rshriram @knrc

**Comment by PiotrSikora** [comment]
Ah, this is kind of blocked on [FilterChain Discovery Service (FDS)](https://github.com/envoyproxy/envoy/issues/4540) and [incremental xDS](https://github.com/envoyproxy/envoy/issues/4991), otherwise each update will require draining of the listener.

**Comment by PiotrSikora** [comment]
Note: Migration to filter chains will require split to separate inbound/outbound listeners (see: #6259), since we need to have different listener filters in both cases (notably, we need TLS Inspector in the inbound listener for the `PERMISSIVE` mode, but we cannot have in the outbound listener).

**Comment by duderino** [comment]
@phildrud how about we give this to @silentdai instead?  He's going to work on the filter chain discovery service in the Envoy-side, so it may make sense for him to do the control plan side too

CC @htuch and @alyssawilk

**Comment by silentdai** [comment]
Working in progress. Currently we cannot move to filter chain due to user impact. If we move migrate the outbound listener to adopt filterchain, any filter chain change will drain the listener and all the filter chains are impacted.
We need Filter Chain Discovery Service, or a subset which restrict the impact of other filter chain.

**Comment by howardjohn** [comment]
@lambdai are we expecting FCDS to be done by 1.4? Seems like this is likely a 1.5 or later item?

**Comment by duderino** [comment]
I think @lambdai should work on this nowish, but I have no opinion on whether 1.4 or 1.5 is feasible

**Comment by lambdai** [comment]
The goal is to be done at 1.4 unless higher priority task reaches. E.g vulnerability...

**Comment by duderino** [comment]
Well stated :)

**Comment by howardjohn** [comment]
Update 6 years later: this is still desired, but low priority

**Comment by howardjohn** [comment]
We will want to implement https://github.com/istio/istio/issues/36596 as part of this
